response_operations_ui/views/messages.py

```diff
@@ -123,12 +125,12 @@ def view_messages():
 
 def _refine(message):
     return {
-        'ru_ref': message.get('ru_id'),
-        'business_name': message.get('@ru_id').get('name'),
+        'ru_ref': _get_ru_ref_from_message(message),
+        'business_name': _get_business_name_from_message(message),
         'subject': message.get('subject'),
         'from': _get_from_name(message),
         'to': _get_to_name(message),
-        'sent_date': message.get('sent_date').split(".")[0]
```

Comment:
given we have switched to the threads view , then its not a given that the latest message retrieved has been sent . It could for example be a draft in which case it would have a DRAFT_SAVED event.